### DISCUSSION TITLE
QR Code Reader files and methods not included when required with composer

### DISCUSSION THREAD

**Discussion by 2blane** [chillerlan/php-qrcode#147] [General]
Hi, I can't call any of the QRCode reader methods like
`$result = (new QRCode($options))->readFromFile(__DIR__.'./eticket.pdf');`
because those methods and files aren't included in the download from Composer. I even looked through the releases and none of those methods like `readFromFile` are included in the download. Is there something I'm missing or a certain way to include the project in composer such that it contains the QRCode reading files? Even when downloading the zip files of the releases none of the reader files are included. Thanks!

**Comment by codemasher** [comment]
That's because it is not released yet. If you want to use the reader you will have to [install it from `dev-main` for now](https://github.com/chillerlan/php-qrcode/wiki/Installation).

**Comment by codemasher** [comment]
Oh on a side note it will probably not read the pdf unless it's somehow supported by ImageMagick.
